lua/telescope/actions/init.lua

```diff
@@ -1463,7 +1463,17 @@ actions.delete_mark = function(prompt_bufnr)
     local bufname = selection.filename
     local bufnr = vim.fn.bufnr(bufname)
     local mark = selection.ordinal:sub(1, 1)
-    local success = pcall(vim.api.nvim_buf_del_mark, bufnr, mark)
+
+    local function is_uppercase_mark(mark)
+      return 65 <= mark:byte() and mark:byte() <= 90
+    end
+
+    local success = false
+    if is_uppercase_mark(mark) then
+      success = pcall(vim.api.nvim_del_mark, mark)
+    else
+      success = pcall(vim.api.nvim_buf_del_mark, bufnr, mark)
+    end
```

Comment:
```suggestion
    local success = false
    if mark:match("%u") then
      success = pcall(vim.api.nvim_del_mark, mark)
    else
      success = pcall(vim.api.nvim_buf_del_mark, bufnr, mark)
    end
```
I think we can simplify this as so